src/os/bluestore/BlueFS.cc

```diff
@@ -479,6 +479,29 @@ void BlueFS::_update_logger_stats()
   }
 }
 
+void BlueFS::collect_read_stalled_alerts(osd_alert_list_t& alerts) {
+  if (!cct->_conf->bluestore_stalled_read_event_counter_threshold) {
+    return;
+  }
+  for (size_t i = 0; i < bdev.size(); ++i) {
+    if (!bdev[i]) continue;
+    bdev[i]->trim_stalled_read_event_queue(mono_clock::now());
+    size_t qsize = bdev[i]->get_stalled_read_event_queue_size();
```

Comment:
You might want to return the queue size from trim_stalled_read_event_queue() and simplify the code here.